profiling/src/internal/profile/fuzz_tests.rs

```diff
@@ -0,0 +1,631 @@
+        }
+    }
+}
+
+impl<'a> From<&'a Location> for api::Location<'a> {
+    fn from(value: &'a Location) -> Self {
+        Self {
+            mapping: (&value.mapping).into(),
+            function: (&value.function).into(),
+            address: value.address,
+            line: value.line,
+        }
+    }
+}
+
+#[derive(Clone, Debug, Default, Eq, PartialEq, Hash, TypeGenerator)]
+pub struct Mapping {
+    /// Address at which the binary (or DLL) is loaded into memory.
+    pub memory_start: u64,
+
+    /// The limit of the address range occupied by this mapping.
+    pub memory_limit: u64,
+
+    /// Offset in the binary that corresponds to the first mapped address.
+    pub file_offset: u64,
+
+    /// The object this entry is loaded from.  This can be a filename on
+    /// disk for the main binary and shared libraries, or virtual
+    /// abstractions like "[vdso]".
+    pub filename: Box<str>,
+
+    /// A string that uniquely identifies a particular program version
+    /// with high probability. E.g., for binaries generated by GNU tools,
+    /// it could be the contents of the .note.gnu.build-id field.
+    pub build_id: Box<str>,
+}
+
+impl Mapping {
+    pub fn new(
+        memory_start: u64,
+        memory_limit: u64,
+        file_offset: u64,
+        filename: Box<str>,
+        build_id: Box<str>,
+    ) -> Self {
+        Self {
+            memory_start,
+            memory_limit,
+            file_offset,
+            filename,
+            build_id,
+        }
+    }
+}
+
+impl<'a> From<&'a Mapping> for api::Mapping<'a> {
+    fn from(value: &'a Mapping) -> Self {
+        Self {
+            memory_start: value.memory_start,
+            memory_limit: value.memory_limit,
+            file_offset: value.file_offset,
+            filename: &value.filename,
+            build_id: &value.build_id,
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct Sample {
+    /// The leaf is at locations[0].
+    pub locations: Vec<Location>,
+
+    /// The type and unit of each value is defined by the corresponding
+    /// entry in Profile.sample_type. All samples must have the same
+    /// number of values, the same as the length of Profile.sample_type.
+    /// When aggregating multiple samples into a single sample, the
+    /// result has a list of values that is the element-wise sum of the
+    /// lists of the originals.
+    pub values: Vec<i64>,
+
+    /// label includes additional context for this sample. It can include
+    /// things like a thread id, allocation size, etc
+    pub labels: Vec<Label>,
+}
+
+impl<'a> From<&'a Sample> for api::Sample<'a> {
+    fn from(value: &'a Sample) -> Self {
+        Self {
+            locations: value.locations.iter().map(api::Location::from).collect(),
+            values: value.values.clone(),
+            labels: value.labels.iter().map(api::Label::from).collect(),
+        }
+    }
+}
+
+/// PartialEq and Hash work on just the key to avoid getting duplicate
```

Comment:
Are we using `Eq` to compare labels in the tests? Will this miss issues where the values don't agree?

Reply:
`std::hash::Hash` doc says this property must hold when implementing `Eq` and `Hash`, `k1 == k2 -> hash(k1) == hash(k2)`. In our case, values can disagree even if the hashes are the same. 

This was introduced mainly to use Labels as keys for HashMap in https://github.com/DataDog/libdatadog/blob/dsn/fuzz-profile-api/profiling/src/internal/profile/fuzz_tests.rs#L369